playerbot/PlayerbotMgr.cpp

```diff
@@ -47,25 +63,31 @@ void PlayerbotHolder::UpdateSessions(uint32 elapsed)
             LogoutPlayerBot(bot->GetObjectGuid().GetRawValue());
         }
     }
+
+   Cleanup();
+}
+
+void PlayerbotHolder::Cleanup()
+{
+   auto it = playerBots.begin();
+
+   while (it != playerBots.end())
+   {
+      if (it->second == nullptr)
+      {
+         it = playerBots.erase(it);
+      }
+
+      ++it;
+   }
 }
 
 void PlayerbotHolder::LogoutAllBots()
 {    
-    /*
-    while (true)
-    {
-        PlayerBotMap::const_iterator itr = GetPlayerBotsBegin();
-        if (itr == GetPlayerBotsEnd()) break;
-        Player* bot= itr->second;
-        if (!bot->GetPlayerbotAI()->isRealPlayer())
-            LogoutPlayerBot(bot->GetObjectGuid().GetRawValue());
-    }
-    */    
-    PlayerBotMap bots = playerBots;
-
-    for (auto& itr : bots)
+    for (auto& itr : playerBots)

```

Comment:
Shouldn't you be using the method you created earlier to iterate between all bots?

Reply:
the idea for the exposed method is not for internal use